kafka-admin/src/test/java/org/bf2/admin/security/TrustedSSOCallerPrincipalFactoryProducerTest.java

```diff
@@ -0,0 +1,122 @@
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import java.security.KeyStore;
+import java.security.cert.CertificateEncodingException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Deprecated(forRemoval = true)
+class TrustedSSOCallerPrincipalFactoryProducerTest {
+
+    @Test
+    void testInitializeHttpsJwksWithValidCert() throws Exception {
+        JWTAuthContextInfo authContextInfo = new JWTAuthContextInfo();
+        authContextInfo.setJwksRefreshInterval(1);
+        TrustedSSOKeyLocationResolver target = new TrustedSSOKeyLocationResolver(authContextInfo) {
+            @Override
+            protected void initializeKeyContent() throws Exception {
+                // No-op
+            }
+        };
+
+        System.setProperty(KafkaAdminConfigRetriever.OAUTH_TRUSTED_CERT, getValidCertificate());
+        TestHttpsJwks httpsJwks;
+
+        try {
+            httpsJwks = (TestHttpsJwks) target.initializeHttpsJwks(new TestHttpsJwks("https://dummy"));
+        } finally {
+            System.clearProperty(KafkaAdminConfigRetriever.OAUTH_TRUSTED_CERT);
+        }
+
+        assertEquals(1, httpsJwks.setSimpleHttpGetInvocationCount);
+    }
+
+    @Test
+    void testInitializeHttpsJwksWithInvalidCert() throws Exception {
+        JWTAuthContextInfo authContextInfo = new JWTAuthContextInfo();
+        authContextInfo.setJwksRefreshInterval(1);
+        TrustedSSOKeyLocationResolver target = new TrustedSSOKeyLocationResolver(authContextInfo) {
+            @Override
+            protected void initializeKeyContent() throws Exception {
+                // No-op
+            }
+        };
+        System.setProperty(KafkaAdminConfigRetriever.OAUTH_TRUSTED_CERT, "INVALID");
+        TestHttpsJwks httpsJwks;
+
+        try {
+            httpsJwks = (TestHttpsJwks) target.initializeHttpsJwks(new TestHttpsJwks("https://dummy"));
+        } finally {
+            System.clearProperty(KafkaAdminConfigRetriever.OAUTH_TRUSTED_CERT);
+        }
+
+        assertEquals(0, httpsJwks.setSimpleHttpGetInvocationCount);
+    }
+
+    @Test
+    void testInitializeHttpsJwksWithoutCert() throws Exception {
+        JWTAuthContextInfo authContextInfo = new JWTAuthContextInfo();
+        authContextInfo.setJwksRefreshInterval(1);
+        TrustedSSOKeyLocationResolver target = new TrustedSSOKeyLocationResolver(authContextInfo) {
+            @Override
+            protected void initializeKeyContent() throws Exception {
+                // No-op
+            }
+        };
+        TestHttpsJwks httpsJwks;
+        httpsJwks = (TestHttpsJwks) target.initializeHttpsJwks(new TestHttpsJwks("https://dummy"));
+        assertEquals(0, httpsJwks.setSimpleHttpGetInvocationCount);
+    }
+
+    static class TestHttpsJwks extends HttpsJwks {
+        int setSimpleHttpGetInvocationCount = 0;
```

Comment:
oddly named member variable

Reply:
The variable is to count the times `setSimpleHttpGet` is invoked :-) Since it's the only method used, I'll remove the method name from the counter to make this easier to read.

Reply:
Updated